api/src/services/usergroup/usergroup.test.ts

```diff
@@ -2,52 +2,284 @@ import { GROUP_ROLE } from '@locokit/lck-glossary'
-    user = await app.service('user').create({
-      name: 'user group test',
-      email: 'hello1@locokit.io',
-      password: 'yo',
-    })
+
+    /**
+     * Create a workspace with default user and authentication
+     */
+    setupData = await builder.setupWorkspace()
+    lambdaUserParams = {
+      provider: 'external',
+      user: setupData.user2,
+      accessToken: setupData.user2Authentication.accessToken,
+      authenticated: true,
+    }
+    adminUserParams = {
+      provider: 'external',
+      user: setupData.userAdmin,
+      accessToken: setupData.userAdminAuthentication.accessToken,
+      authenticated: true,
+    }
+    superAdminUserParams = {
+      provider: 'external',
+      user: setupData.userSuperAdmin,
+      accessToken: setupData.userSuperAdminAuthentication.accessToken,
+      authenticated: true,
+    }
   })
+
   it('registered the service', () => {
-    expect(service).toBeTruthy()
+    expect(app.service('usergroup')).toBeTruthy()
   })
 
   it('throw if user_id doesn\'t exist', async () => {
     expect.assertions(1)
-    await expect(service.create({
+    await expect(app.service('usergroup').create({
       group_id: group.id,
     })).rejects.toThrow()
   })
+
   it('throw if group_id doesn\'t exist', async () => {
     expect.assertions(1)
-    await expect(service.create({
-      user_id: user.id,
+    await expect(app.service('usergroup').create({
+      user_id: setupData.user2.id,
     })).rejects.toThrow()
   })
-  it('throw if group_id doesn\'t exist', async () => {
+
+  it('add existing user in an existing group', async () => {
     expect.assertions(3)
-    const uhg = await service.create({
-      user_id: user.id,
+    const uhg = await app.service('usergroup').create({
+      user_id: setupData.user2.id,
       group_id: group.id,
     })
     expect(uhg.uhg_role).toBe(GROUP_ROLE.MEMBER)
-    expect(uhg.user_id).toBe(user.id)
+    expect(uhg.user_id).toBe(setupData.user2.id)
     expect(uhg.group_id).toBe(group.id)
-    await service.remove(`${uhg.user_id as string},${uhg.group_id as string}`)
+    await app.service('usergroup').remove(`${uhg.user_id as number},${uhg.group_id as string}`)
+  })
+
+  it("prevent a user with USER's role to assign user to group", async () => {
+    expect.assertions(1)
+
+    const user = await app.service('user').create({
+      email: 'originalUser@locokit.io',
+      name: 'testing create user',
+    }) as LckUser
+
+    const userGroup = await expect(app.service('usergroup').create({
+      user_id: user.id,
+      group_id: group.id,
+    }, lambdaUserParams)).rejects.toThrowError(MethodNotAllowed) as Usergroup | undefined
+    // Clean the database whether the test succeeds or not
+    if (userGroup) await app.service('usergroup').remove(`${userGroup.user_id as number},${userGroup.group_id as string}`)
+    if (user) await app.service('user').remove(user.id)
+  })
+
+  it("prevent a user with USER's role to modify user in a group", async () => {
```

Comment:
This is wrong, because if the user have a `USER` profile, but is a `ADMIN`/`OWNER` of the group, or a member of a `manager` aclset, he could do that ?